src/node-pool/index.ts

```diff
@@ -0,0 +1,164 @@
+
+  init ({ nodes = [] }: {nodes: Node[]}): void {
+    this.pool = new Map()
+    nodes.forEach((node, i: number) => {
+      const { name, instance } = node
+      this.addNode(name, instance, i === 0)
+    })
+
+    if (nodes.length > 0) this.selectNode(nodes[0].name)
+
+    Object.defineProperties(this, {
+      api: {
+        enumerable: true,
+        configurable: false,
+        get () {
+          if (this.selectedNode == null || this.selectedNode.instance == null) {
+            throw new NodeNotFoundError('You can\'t use Node API. Node is not connected or not defined!')
+          }
+          return this.selectedNode.instance.api
+        }
+      }
+    })
+  }
+
+  /**
+     * Add Node
+     * @alias module:@aeternity/aepp-sdk/es/node-pool
+     * @param name - Node name
+     * @param node - Node instance
+     * @param select - Select this node as current
+     * @example
+     * // add and select new node with name 'testNode'
+     * nodePool.addNode('testNode', awaitNode({ url }), true)
+     */
+  addNode (name: string, node: Node, select = false): void {
+    if (this.pool.has(name)) throw new DuplicateNodeError(name)
+
+    this.pool.set(name, {
+      name,
+      instance: node,
+      url: node.url,
+      networkId: node.nodeNetworkId,
+      version: node.version,
+      consensusProtocolVersion: node.consensusProtocolVersion
+    })
+    if (select || this.selectedNode == null) {
+      this.selectNode(name)
+    }
+  }
+
+  /**
+     * Select Node
+     * @alias module:@aeternity/aepp-sdk/es/node-pool
+     * @param name - Node name
+     * @example
+     * nodePool.selectNode('testNode')
+     */
+  selectNode (name: string): void {
+    if (!this.pool.has(name)) throw new NodeNotFoundError(`Node with name ${name} not in pool`)
+    this.selectedNode = this.pool.get(name)
+  }
+
+  /**
+     * Get NetworkId of current Node
+     * @alias module:@aeternity/aepp-sdk/es/node-pool
+     * @example
+     * nodePool.getNetworkId()
+     */
+  getNetworkId: (node?: any) => string = getNetworkId
+
+  /**
+     * Check if you have selected node
+     * @alias module:@aeternity/aepp-sdk/es/node-pool
+     * @example
+     * nodePool.isNodeConnected()
+     */
+  isNodeConnected (): boolean {
+    return this.selectedNode.instance != null
+  }
+
+  /**
+     * Get information about node
+     * @alias module:@aeternity/aepp-sdk/es/node-pool
+     * @example
+     * nodePool.getNodeInfo() // { name, version, networkId, protocol, ... }
+     */
+  getNodeInfo (): NodeInfo {
+    if (!this.isNodeConnected()) throw new DisconnectedError()
+    return {
+      name: this.selectedNode.name,
+      ...this.selectedNode.instance.getNodeInfo()
+    }
+  }
+
+  /**
+     * Get array of available nodes
+     * @alias module:@aeternity/aepp-sdk/es/node-pool
+     * @example
+     * nodePool.getNodesInPool()
+     */
+  getNodesInPool (): NodeInfo[] {
+    return Array.from(this.pool.entries()).map(([name, node]: [string, Node]) => ({
```

Comment:
```suggestion
    return Array.from(this.pool.entries()).map(([name, node]) => ({
```
we can omit type in this case, because it can be derived from the pool type (also linter don't forces us to do that)